File: tests/transpiler/x/pas/left_join.pas
{$mode objfpc}
program Main;
type Anon38 = record
  id: integer;
  name: string;
end;
type Anon39 = record
  id: integer;
  customerId: integer;
  total: integer;
end;
type Anon40 = record
  orderId: integer;
  customer: Anon38;
  total: integer;
end;
var
  customers: array of Anon38;
  orders: array of Anon39;
  matched2: boolean;
  result: array of Anon40;
  o: Anon39;
  c: Anon38;
  entry: integer;
begin
  customers := [(id: 1; name: 'Alice'), (id: 2; name: 'Bob')];
  orders := [(id: 100; customerId: 1; total: 250), (id: 101; customerId: 3; total: 80)];
  result := [];
  for o in orders do begin
  matched2 := false;
  for c in customers do begin
  if o.customerId = c.id then begin
  result := concat(result, [(orderId: o.id; customer: (id: 0; name: ''); total: o.total)]);
  matched2 := true;
end;
end;
  if not matched2 then begin
  result := concat(result, [(orderId: o.id; customer: (id: 0; name: ''); total: o.total)]);
end;
end;
  writeln('--- Left Join ---');
  for entry in result do begin
  writeln('Order', entry.orderId, 'customer', entry.customer, 'total', entry.total);
end;
end.

File: archived/tests/transpiler/x/pl/left_join.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    Customers = [_{id: 1, name: "Alice"}, _{id: 2, name: "Bob"}],
    Orders = [_{id: 100, customerId: 1, total: 250}, _{id: 101, customerId: 3, total: 80}],
    Result = [_{orderId: 100, customer: _{id: 1, name: "Alice"}, total: 250}, _{orderId: 101, customer: _{}, total: 80}],
    writeln("--- Left Join ---"),
    Entry = _{orderId: 100, customer: _{id: 1, name: "Alice"}, total: 250},
write("Order"), write(' '), write(100), write(' '), write("customer"), write(' '), write(_{id: 1, name: "Alice"}), write(' '), write("total"), write(' '), writeln(250),
    Entry1 = _{orderId: 101, customer: _{}, total: 80},
write("Order"), write(' '), write(101), write(' '), write("customer"), write(' '), write(_{}), write(' '), write("total"), write(' '), writeln(80).
